src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -784,10 +783,14 @@ def _generate_properties(api_version, orchestrator_type, orchestrator_version, m
     return properties
 
 
-def _get_user_assigned_identity_client_id(cli_ctx, resource_id):
-    pattern = '/subscriptions/(.*?)/resourcegroups/(.*?)/providers/microsoft.managedidentity/userassignedidentities/(.*)'  # pylint: disable=line-too-long
+_re_user_assigned_identity_resource_id = re.compile(
```

Comment:
Should we try to avoid global variables like ```_re_user_assigned_identity_resource_id```? Better to wrap this with a function, or move it to _consts.py as a constant.

Reply:
Makes sense, updated.